build_scripts/build.sh

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env bash
+
+# The MIT License (MIT)
+
+# Copyright 2022 Hewlett Packard Enterprise Development LP.
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+VENV_PATH="/tmp/poetryvenv"
```

Comment:
That's a good point. Part of the issue here is that `poetry` lives in its own virtualenv and then creates its own virtualenvs for the each managed project, in this case `csm-api-client` itself. I suppose creating a project-specific Poetry virtualenv couldn't hurt, though. It would at least make it explicit what the new virtualenv is for.

Reply:
>`poetry` lives in its own virtualenv and then creates its own virtualenvs for the each managed project

Ah OK, I did not realize this when I wrote that comment. I still think I'd probably prefer the path be relative to the current directory if possible.

Sorry to make you explain all of poetry to me in this PR, but when you run `build.sh prep` and it runs `in_venv poetry install`, isn't this still installing `csm-api-client` and its dependencies into `/tmp/poetryvenv`? That's where I'm wondering about having separate venvs.

Reply:
When you run `poetry install`, it installs the dependencies of the project described by `pyproject.toml` into its own virtualenv which Poetry creates, separate from the virtualenv which Poetry is installed in (i.e. `/tmp/poetryvenv`).

To illustrate, consider the following function added to `build.sh`:
```bash
function show_virtualenv() {
    in_venv poetry env info
}
```
Running `./build.sh prep && ./build show_virtualenv` shows the following:
```
Virtualenv
Python:         3.8.10
Implementation: CPython
Path:           /Users/jstanek/Library/Caches/pypoetry/virtualenvs/csm-api-client-I3FZxQ54-py3.8
Executable:     /Users/jstanek/Library/Caches/pypoetry/virtualenvs/csm-api-client-I3FZxQ54-py3.8/bin/python
Valid:          True

System
Platform:   darwin
OS:         posix
Python:     3.8.10
Path:       /Users/jstanek/.pyenv/versions/3.8.10
Executable: /Users/jstanek/.pyenv/versions/3.8.10/bin/python3.8
```

Reply:
That's really helpful, thanks. I think this is probably related to the other comment thread about my unit test issues, I think I'm just having some problems on my end.

Reply:
Actually, disregard the comment above please. It was installing into `/tmp/poetryvenv` specifically because I'd modified `build.sh` to set `$VIRTUAL_ENV` - according to the poetry docs if you're already in a virtualenv it won't create a new one: 
> If Poetry detects it’s running within an activated virtual environment, it will never create a new virtual environment, regardless of the value set for virtualenvs.create.
https://python-poetry.org/docs/configuration/#virtualenvscreate